src/DIRAC/Core/Base/DIRACDB.py

```diff
@@ -14,7 +14,10 @@ def __init__(self, *args, **kwargs):
 
         :param self: self reference
         """
-        self.log = gLogger.getSubLogger(self.fullname)  # pylint: disable=no-member
+        parentLogger = kwargs.pop("parentLogger", None)
+        if not parentLogger:
+            parentLogger = gLogger
+        self.log = parentLogger.getSubLogger(self.fullname)  # pylint: disable=no-member
```

Comment:
Should this emit a warning? (And possibly respect `DIRAC_DEPRECATED_FAIL`.)

Reply:
I don't think so, I prefer to keep it backward compatible. Moreover, there are cases where it does not work (the RSS DBs are non sense but I managed to pass it their logger anyway, but the MonitoringDB is a desperate case). 
Also, it would be a real burden in tests